libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -432,6 +433,15 @@ public void onInstructionListVisibilityChanged(boolean visible) {
     });
   }
 
+  private void updateSavedInstanceStateMapStyle(@NonNull MapView mapView, @Nullable Bundle savedInstanceState) {
+    String mapStyleUrl = ThemeSwitcher.retrieveMapStyle(getContext());
+    if (savedInstanceState != null) {
+      savedInstanceState.putString(MapboxConstants.STATE_STYLE_URL, mapStyleUrl);
+    } else {
+      mapView.setStyleUrl(mapStyleUrl);
```

Comment:
@Guardiola31337 found the issue - we were previously only updated when the saved instance state _was not_ null, thus it was always day mode style.